src/tools/iCubSkinGui/plugin/include/SkinMeshThreadCan.h

```diff
@@ -141,6 +163,14 @@ class SkinMeshThreadCan : public PeriodicThread
                         if (type=="fingertip3R")
                         {
                             sensor[id]=new Fingertip3R(xc,yc,th,gain,layoutNum,lrMirror);
+                        }
+                         if (type=="fingertip4R")
+                        {
+                            sensor[id]=new Fingertip4R(xc,yc,th,gain,layoutNum,lrMirror);
+                        }
+                         if (type=="fingertipMID")
+                        {
+                            sensor[id]=new FingertipMID(xc,yc,th,gain,layoutNum,lrMirror);
```

Comment:
Better off turning this check that is getting too long into a chain of `else-if` as already done elsewhere within this codebase.

Reply:
2202a362d7f76fd55126b9505dcd116b541d1d50 resolved both comments here and above

Reply:
Thanks!